airsonic-main/src/main/java/org/airsonic/player/service/TranscodingService.java

```diff
@@ -485,6 +487,59 @@ private boolean isTranscodingStepInstalled(String step) {
         return matches != null && matches.length > 0;
     }
 
+    /**
+     * Returns the length (or predicted/expected length) of a (possibly padded) media stream
+     */
+    private Long getExpectedLength(Parameters parameters) {
+        MediaFile file = parameters.getMediaFile();
+
+        if (!parameters.isDownsample() && !parameters.isTranscode()) {
+            return file.getFileSize();
+        }
+        Integer duration = file.getDurationSeconds();
+        Integer maxBitRate = parameters.getMaxBitRate();
+
+        if (duration == null) {
+            LOG.warn("Unknown duration for " + file + ". Unable to estimate transcoded size.");
+            return null;
+        }
+
+        if (maxBitRate == null) {
+            LOG.error("Unknown bit rate for " + file + ". Unable to estimate transcoded size.");
+            return null;
+        }
+
+        // Over-estimate size a bit (2 seconds) so don't cut off early in case of small calculation differences
+        return (duration + 2) * (long)maxBitRate * 1000L / 8L;
```

Comment:
Please don't use magic constants like `1000L / 8L` without at least adding a comment :)

Reply:
That function was just moved from `StreamController`, I didn't write it... :-)